### PR TITLE
[EOC-89] Use common dialog in project

### PR DESCRIPTION
**Pull request by adamklepacz**
CreationForm and ModalForm removed and replaced by 
`<Dialog />` and` <FormDialog />.`
`<DropdownForm />` modified to user` <Form/> `instead of `<CreationForm />.`

I have also changed props name on the <Form /> to match convention from an article that Marek sent.
I've got some news that dropdowns will be replaced for dialogs in the future. So this implementation is just for the current use case.

### CODE REVIEW

File: client/src/common/components/Form.jsx [hunk 1]
```diff
@@ -76,8 +76,8 @@ Form.propTypes = {
   defaultDescription: PropTypes.string,
   defaultName: PropTypes.string,
 
-  handleDescriptionChange: PropTypes.func.isRequired,
-  handleNameChange: PropTypes.func.isRequired,
+  onDescriptionChange: PropTypes.func.isRequired,
+  onNameChange: PropTypes.func.isRequired,
   onSubmit: PropTypes.func
```
Comment (alemikolo) [inline client/src/common/components/Form.jsx]: whether onSubmit shouldn't be require?

File: client/src/modules/shopping-list/index.jsx [hunk 2]
```diff
@@ -139,19 +139,19 @@ class ShoppingList extends Component {
...
             onConfirm={this.archiveListHandler(listId)}
           />
         )}
         {showUpdateForm && (
-          <ModalForm
+          <FormDialog
             defaultDescription={description}
             defaultName={name}
-            label="Edit list"
+            title="Edit list"
```
Comment (alemikolo) [inline client/src/modules/shopping-list/index.jsx]: alphabetic order

File: client/src/modules/shopping-list/index.jsx [hunk 3]
```diff
@@ -139,19 +139,19 @@ class ShoppingList extends Component {
         )}
         {isArchived && <ArchivedList listId={listId} name={name} />}
         {showDialogBox && (
-          <DialogBox
-            message={`Do you really want to archive the ${name} list?`}
+          <Dialog
+            title={`Do you really want to archive the ${name} list?`}
```
Comment (alemikolo) [inline client/src/modules/shopping-list/index.jsx]: alphabetic order

File: client/src/common/components/DropdownForm.jsx [hunk 4]
```diff
@@ -1,22 +1,81 @@
...
+              </button>
+            </div>
+          </div>
+          <Overlay onClick={onHide} type={OverlayStyleType.LIGHT} />
+        </Fragment>
+      )
+    );
+  }
+}
 DropdownForm.propTypes = {
```
Comment (marekrozmus) [inline client/src/common/components/DropdownForm.jsx]: new line missing

File: client/src/common/components/Form.jsx [hunk 5]
```diff
@@ -16,20 +16,20 @@ class Form extends PureComponent {
     const {
       target: { value }
     } = event;
-    const { handleNameChange } = this.props;
+    const { onNameChange } = this.props;
 
     this.setState({ name: value });
-    handleNameChange(value);
+    onNameChange && onNameChange(value);
```
Comment (marekrozmus) [inline client/src/common/components/Form.jsx]: thos one is required in props so no need to check it here - you'll get error about missing props during component creation

File: client/src/common/components/Form.jsx [hunk 6]
```diff
@@ -16,20 +16,20 @@ class Form extends PureComponent {
...
   handleDescriptionChange = event => {
     const {
       target: { value }
     } = event;
-    const { handleDescriptionChange } = this.props;
+    const { onDescriptionChange } = this.props;
 
     this.setState({ description: value });
-    handleDescriptionChange(value);
+    onDescriptionChange && onDescriptionChange(value);
```
Comment (marekrozmus) [inline client/src/common/components/Form.jsx]: thos one is required in props so no need to check it here - you'll get error about missing props during component creation

**Pull request merged**